### PR TITLE
Improve Shared Interface documentation

### PR DESCRIPTION
**Pull request by rlouedec-microej**

### CODE REVIEW

**Comment by rlouedec-microej** [comment]
Links to the Shared Interface examples repository will be added when it is published.

File: KernelDeveloperGuide/featuresCommunication.rst [hunk 1]
```diff
@@ -6,6 +6,35 @@ Communication between Features
 Features can communicate together through the use of Shared Interfaces.
 This mechanism is described in :ref:`chapter.shared.interfaces` section.
 
+For Applications to use the Shared Interface mechanism, a Kernel must provide:
+
+* a KF service registry
+* a list of Kernel types converters
+
```
Comment (frivieremicroej) [inline KernelDeveloperGuide/featuresCommunication.rst:13]:
```suggestion
* an API for a first Application to register its Share Interface, and for a second Application to get a proxy on it
* a set of registered Kernel types converters
```

**Review by frivieremicroej — CHANGES_REQUESTED** [review]

File: ApplicationDeveloperGuide/sandboxedAppSharedInterface.rst [hunk 2]
```diff
@@ -32,58 +34,178 @@ The basic schema:
...
+   #. Register the instance to a KF service registry
+#. From the consumer application,
 
--  Interface registration.
+   #. Retrieve a proxy of the instance from the KF service registry
+   #. Call methods of the instance proxy.
 
-Interface Definition
-~~~~~~~~~~~~~~~~~~~~
+Shared Interface Definition
```
[the remark this answers is not in the document]
Comment (rlouedec-microej) [reply ApplicationDeveloperGuide/sandboxedAppSharedInterface.rst]: Done in: 43ba03c7459c6088a618ae070fe5216a715ddf10

Should I apply this to more section titles ?

[the remark this answers is not in the document]
Comment (rlouedec-microej) [reply ApplicationDeveloperGuide/sandboxedAppSharedInterface.rst]: Applied in: 43ba03c7459c6088a618ae070fe5216a715ddf10

**Comment by frivieremicroej** [comment]
@rlouedec-microej updated & approved this MR.
The documentation is up-to-date with the current version of the GREEN Firmware. 
It will be need to be updated after the registry usage refactor (https://youtrack.cross/issue/J0095K-1404/Switch-from-Wadapps-shared-registry-to-ejservice)

**Pull request merged**